lua/lspconfig/server_configurations/rust_analyzer.lua

```diff
@@ -1,13 +1,16 @@
 local util = require 'lspconfig.util'
 
-local function reload_workspace(bufnr)
-  bufnr = util.validate_bufnr(bufnr)
-  vim.lsp.buf_request(bufnr, 'rust-analyzer/reloadWorkspace', nil, function(err)
-    if err then
-      error(tostring(err))
-    end
-    vim.notify 'Cargo workspace reloaded'
-  end)
+local function reload_workspace()
+  local client = vim.lsp.get_active_clients { name = 'rust_analyzer' }
+  if #client ~= 0 then
+    vim.notify 'Reloading Cargo Workspace'
+    client[1].request('rust-analyzer/reloadWorkspace', nil, function(err)
```

Comment:
there can't use 1 index if there has two ?   get client which attach current buffer?

Reply:
I’m using index 1 because I’m searching clients by name. It’s possible for multiple clients to attach to one buffer, but if there are multiple clients with the name `”rust_analyzer”` then something has clearly gone wrong.

Reply:
> but if there are multiple clients with the name ”rust_analyzer” then something has clearly gone wrong.

maybe nope. if someone rewrite root detect for spawn two servers? Please don't use hard-coded assumptions. It's not 100% established

Reply:
Alright, will update accordingly.